tests/integration/Core/Checkout/Promotion/Cart/PromotionCalculatorTest.php

```diff
@@ -151,12 +151,14 @@ public function testAutomaticExclusionsDontAddError(): void
         $firstPromotionId = $this->getPromotionId(true, 1, false);
         $discountItemToBeExcluded = $this->getDiscountItem($firstPromotionId);
         $discountItemToBeExcluded->setPayloadValue('preventCombination', true);
+        $discountItemToBeExcluded->setPayloadValue('priority', 1);
 
         $secondPromotionId = $this->getPromotionId(true, 2, false);
         $validDiscountItem = $this->getDiscountItem($secondPromotionId);
         $validDiscountItem->setPayloadValue('preventCombination', true);
+        $validDiscountItem->setPayloadValue('priority', 2);
```

Comment:
Can you add a testcase as well that the behaviour actually differs when you switch the priority, so that we can assure that the priority is really taken into account correctly?

Reply:
Sure, I've adjusted the testcase to check for correct behaviour when switching priorities.